controllers/fenceagentsremediation_controller.go

```diff
@@ -167,9 +167,18 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 		}
 
 		// remove node's taints
-		if err := utils.RemoveTaint(r.Client, far.Name); err != nil && !apiErrors.IsNotFound(err) {
-			return emptyResult, err
+		taint := utils.CreateRemediationTaint()
+		if err := utils.RemoveTaint(r.Client, far.Name, taint); err != nil {
+			if apiErrors.IsConflict(err) {
+				r.Log.Info("Failed to remove taint from node due to node update, retrying... ,", "node name", node.Name, "taint key", taint.Key, "taint effect", taint.Effect)
+				return ctrl.Result{RequeueAfter: time.Second}, nil
```

Comment:
Why requeue after one second and not immediately afterwards since this conflict should not happen multiple times, right?

Reply:
Giving the update a bit of time to finish, no point in stressing the system and spamming the log

Reply:
Well, in case we want to utilize the exponential back-off mechanism we must return an error which will also be logged as such.
IIUC it is something we want to avoid (logging an error) since this behavior is not uncommon and expected .

Reply:
`Requeue: True` also results in rate limited requeue
https://github.com/kubernetes-sigs/controller-runtime/blob/56159419231e985c091ef3e7a8a3dee40ddf1d73/pkg/internal/controller/controller.go#L341

Reply:
>Requeue: True also results in rate limited requeue

I agree this is what I originally meant. But IIRC Michael suspects that this error is recurrent (I didn't see a proof for that), it seems more like a temporary error to me.